internal/controller/preservation/helper_test.go

```diff
@@ -49,71 +49,93 @@ func ensureEquality(actual, expected client.Object) {
+			Expect(err.Error()).To(ContainSubstring("unknown PreservationMode"))
+			Expect(lSelector).To(BeNil())
+		})
 
-			It("returns a label.selector for retrieving cluster identity preservation resources when preservationMode is ClusterIdentity", func() {
-				var (
-					expected *labels.Requirement
-					err      error
-				)
-				if expectedLabelValues == nil {
-					expected, err = labels.NewRequirement(validPreservationLabelKey, selection.Exists, nil)
-				} else {
-					expected, err = labels.NewRequirement(validPreservationLabelKey, selection.Equals, expectedLabelValues)
-				}
+		It("returns a nil label.selector when preservationMode is None", func() {
+			verifyLabelSelectorFn(v1alpha1.PreservationModeNone, labelSelectorBuilderFn, BeNil())
+		})
 
-				Expect(err).NotTo(HaveOccurred())
-				verifyLabelSelectorFn(v1alpha1.PreservationModeClusterIdentity, Equal(labels.NewSelector().Add(*expected)))
-			})
+		It("returns a label.selector for retrieving all preservation resources when preservationMode is All", func() {
+			expected, err := labels.NewRequirement(v1alpha1.PreservationLabelKey, selection.Exists, nil)
+			Expect(err).NotTo(HaveOccurred())
+			verifyLabelSelectorFn(v1alpha1.PreservationModeAll, labelSelectorBuilderFn, Equal(labels.NewSelector().Add(*expected)))
+		})
 
+		It("returns a label.selector for retrieving all preservation resources when preservationMode is ClusterIdentity", func() {
+			expected, err := labels.NewRequirement(v1alpha1.PreservationLabelKey, selection.Equals, []string{v1alpha1.ClusterIdentityLabelValue})
+			Expect(err).NotTo(HaveOccurred())
+			verifyLabelSelectorFn(v1alpha1.PreservationModeClusterIdentity, labelSelectorBuilderFn, Equal(labels.NewSelector().Add(*expected)))
 		})
 
-		Context("preservationMode is invalid", func() {
-			It("errors and returns a nil label.selector", func() {
-				var mode v1alpha1.PreservationMode = "foobar"
-				lSelector, err := labelSelectorBuilderFn(mode)
-				Expect(err).To(HaveOccurred())
-				Expect(err.Error()).To(ContainSubstring("unknown PreservationMode"))
-				Expect(lSelector).To(BeNil())
-			})
+		It("returns a label.selector for retrieving all preservation resources when preservationMode is Internal", func() {
+			expected, err := labels.NewRequirement(InternalPreservationLabelKey, selection.Equals, []string{InternalPreservationLabelValue})
+			Expect(err).NotTo(HaveOccurred())
+			verifyLabelSelectorFn(PreservationModeInternal, labelSelectorBuilderFn, Equal(labels.NewSelector().Add(*expected)))
 		})
-	}
 
-	Describe("buildBackupLabelSelector", func() {
-		commonTests(buildBackupLabelSelector, v1alpha1.PreservationLabelKey, []string{v1alpha1.ClusterIdentityLabelValue})
 	})
 
 	Describe("buildRestoreLabelSelector", func() {
-		commonTests(buildRestoreLabelSelector, preservedDataLabelKey, nil)
+
+		labelSelectorBuilderFn := buildRestoreLabelSelector
+
+		It("errors and returns a nil label.selector for unknown preservationMode", func() {
```

Comment:
Why do all of these say "label.selector"? I feel like that's a strange way to write that. Don't most places write it like `label selector`?

Reply:
I've used `label.selector` throughout the test file - maintaining consistency.

Reply:
:shrug: sure